src/pybind/mgr/volumes/fs/async_job.py

```diff
@@ -251,7 +251,8 @@ def _cancel_job(self, volname, job):
         canceled = False
         log.info("canceling job {0} for volume {1}".format(job, volname))
         try:
-            if not volname in self.q and not volname in self.jobs and not job in self.jobs[volname]:
+            vol_jobs = [j[0] for j in self.jobs[volname]]
+            if not volname in self.q and not volname in self.jobs and not job in vol_jobs:
```

Comment:
Also fixed this check.

Reply:
You'd to make this change as currently we allow multiple cloner threads per volume?

Reply:
No, the conditional check was logically incorrect. 

job = (trackid, clone_base_path)
self.jobs[volname] = [((trackid, clone_base_path), thread_id)]

So we need construct the list with only the first element in the tuple leaving out thread_id to check for the job.

Reply:
Thanks for the explanation.